### ISSUE TITLE
How to pass Firefox settings

### ISSUE THREAD

**Issue by zoechi** [google/webdriver.dart#98]
I failed miserable trying to figure out how to pass something like `'reader.parse-on-load.enabled': false` to Firefox using `desired`. There are examples how to use Java classes (like https://github.com/SeleniumHQ/selenium/blob/master/java/client/src/org/openqa/selenium/firefox/FirefoxProfile.java), but I wasn't yet able to figure out how to do this with webdriver.dart. 
Any hint would be appreciated.

**Comment by DrMarcII** [comment]
How Selenium handles Firefox profiles is exceedingly ugly, and falls well into the category of things I don't want to implement. It is also difficult to implement in Dart as it requires zip'ing a directory of files and there isn't a Dart zip library that I know of.

That said, if you wanted to create a separate library for encoding a Firefox profile that calls out to command-line zip, I would not be opposed to adding it the support directory. The basic idea would be to create a directory on disk that conforms to whatever a Firefox profile is supposed to look like (this will require significant reverse-engineering of the code for one of the Selenium clients). Then zip that directory with command-line zip, read in the resulting zip file and send it in the appropriate field of the capabilities as a BASE64-encoded string.

**Comment by zoechi** [comment]
Thanks, that explains a lot! I thought it only needs to be added to `desired` as JSON and passed to webdriver as such. Now I'm already less confused about the Java code. There are zip libraries available for Dart (with some limitations). I'll have a look.

**Comment by zoechi** [comment]
Now that I read your last message again, in the end it's still just sending data in a field of `desired`, as I assumed, right? Just getting the right data in the required format is the hard work.

**Comment by zoechi** [comment]
I got it working \o/

Summary for others who might also need it:
I created a zip of a profile directory manually and passed its content base64-encoded to desired:

```dart
import 'package:crypto/crypto.dart';
...
var firefoxProfile = CryptoUtils.bytesToBase64(
    new io.File('/home/myuser/.mozilla/firefox/webdriver_profile.zip').readAsBytesSync());
var driver = createDriver(uri: uri, 
    desired: {'browserName': Browser.firefox, 'firefox_profile': firefoxProfile})`.
```
The zip must only contain the files and subdirectories of the profile directory but not the profile directory itself (no root directory in the zip).
The zip needs to contain at least the file `prefs.js` it seems. 
I got the desired Firefox behavior by just adding the `prefs.js` (auto generated when Firefox is started with a new profile) and my custom settings in `user.js` and no further files. Passing the whole set of files generated in the Firefox profile directory with my `user.js` worked as well.

Thanks a lot for your help! 

Maybe I'll try to create something more convenient and reach out to you about integrating it into this project.
